woocommerce/woocommerce-ios
# Image Upload: loading state UI and integration

"UI (Product image loading state) and integration with MediaStore" part of #1713 

## Summary

This PR created the UI for Product image loading state, and integrated the UI with image upload functionality.

## Changes

- Created `CircleSpinnerView` to display an animatable circle spinner as in the design spec
- Created `InProgressProductImageCollectionViewCell` cell when an image is still being processed/uploaded
- Created an enum `ProductImageStatus` that indicates the status of a Product image with 2 cases for now: remote image, or uploading an asset
- Created `ProductImagesService` that encapsulates the implementation of uploading an image asset
- In `ProductImagesViewController` and `ProductImagesCollectionViewController`, updated its data source to an array of `ProductImageStatus` instead of `ProductImage` and displayed the cell and image based on the status enum
- In `ProductImagesViewController`, uploaded a media asset with `ProductImagesService` and updated the corresponding cell upon an error or remote `ProductImage`
  - Since the order of the image statuses could change when we implement reordering, it searches the first index path matching the asset

## Testing

Testing on a device is recommeded so that the camera is enabled

- Launch the app
- Go to Products tab
- Tap on a simple Product
- Tap on the + UI in the images header
- Tap "Add Photos"
- Tap "Choose from device"
- Pick a few photos from the photo library
- Tap "Add #" in the navigation bar --> should see the correct number of images added to the images UI with loading state
- Wait for a bit, and feel free to play around like tapping "Done", navigating back, tapping on another image and deleting it --> the image(s) should be uploaded given network connection
- Repeat the above steps with the camera from "Add Photos" action sheet

## Example screenshots

loading UI | image upload
-- | --
<img src="https://user-images.githubusercontent.com/1945542/73436170-f9911380-4384-11ea-9209-6f3354ba2201.PNG" width="600" /> | ![image_upload_v1](https://user-images.githubusercontent.com/1945542/73438649-54c50500-4389-11ea-89b3-e4b0a0a2d297.gif)


Update release notes:
